ls1intum/Artemis
# `Exam mode`: Fix an issue when canceling assessments

### Checklist
#### General
#### Client

### Motivation and Context
Fixes #7277. Canceling text exercise assessments is not possible due to a missing participation value.

### Description
Instead of using `result.participation` (which would be a backreference since the result source is `participation.results[correctionRound]`) we now directly use the participation present in the template.

Note that text exercises are the only exercises that actually need the participation id here. In the future, we should unify these endpoints and also only use the result id here.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Exam with text exercises and second correction enabled.

0. Edit the exam and set the number of correction rounds to 2.
1. Start a text exercise assessment in the second correction round (let one tutor assess the exercise in the first round, active second correction, then let a second tutor start the assessment in the second correction round, but don't submit the assessment yet)
2. Open the scores page (exercise groups -> exercise -> scores) and cancel the assessment
3. Check that no error happens and the assessment got canceled.

### Review Progress

#### Performance Review
#### Code Review
#### Manual Tests
